src/text/span.rs

```diff
@@ -6,22 +6,55 @@ use unicode_width::UnicodeWidthStr;
 use super::StyledGrapheme;
 use crate::style::{Style, Styled};
 
-/// A string where all graphemes have the same style.
+/// A `Span` represents a part of a line that is contiguous and where all characters share the same
+/// style.
+///
+/// It is the smallest unit of text that can be styled. It is used by the [`Line`] type to represent
+/// a line of text where each grapheme has its own style.
+///
+/// # Examples
+///
+/// `Span`a can be created from a `&str`, a `String` or a `Cow<str>`. It can also be created from a
+/// `&str` or a `String` using the `From` trait, which will create a `Span` with no style.
+/// Alternatively, a `Span` can be created from a `&str` or a `String` using the `Stylize` trait,
```

Comment:
All the creation options listed here except for `Cow<str>` seem redundant to the example, so maybe a little wordy considering the example follows right after ?

lmk what you think